src/NearestNeighbors.jl

```diff
@@ -73,6 +73,7 @@ const KNN = Union{KNNRegressor, KNNClassifier}
 function MLJBase.fit(m::KNN, verbosity::Int, X, y, w=nothing)
     if w !== nothing
         length(w) == length(y) || throw(ArgumentError("The weights vector `w` must have the same length as that of the response `y`."))
+        all(w .≥ 0) || throw(ArgumentError("Only non-negative weights are supported."))
     end
```

Comment:
Dimension matching will be checked by machine constructor, so can probably drop this.
The positivity check is not there, but I will add, thanks.